picard/ui/coverartbox.py

```diff
@@ -169,16 +177,29 @@ def fetch_remote_image(self, url):
                     data = f.read()
                 self.load_remote_image(url, mime, data)
 
-    def on_remote_image_fetched(self, url, data, reply, error):
+    def on_remote_image_fetched(self, url, data, reply, error, fallbackData=None):
         mime = reply.header(QtNetwork.QNetworkRequest.ContentTypeHeader)
         if mime in ('image/jpeg', 'image/png'):
             self.load_remote_image(url, mime, data)
-        elif reply.url().hasQueryItem("imgurl"):
+        elif url.hasQueryItem("imgurl"):
             # This may be a google images result, try to get the URL which is encoded in the query
-            url = QtCore.QUrl(reply.url().queryItemValue("imgurl"))
+            url = QtCore.QUrl(url.queryItemValue("imgurl"))
             self.fetch_remote_image(url)
         else:
-            log.warning("Can't load image with MIME-Type %s", mime)
+            log.warning("Can't load remote image with MIME-Type %s", mime)
+            if fallbackData:
+               # Tests for image format obtained from file-magic
+               if fallbackData[:2]=='\xff\xd8':
```

Comment:
Note that fallbackData is a QByteArray object, so it doesn't have a startswith method, but startsWith. In fact, I was tempted to use startsWith but then thought using [:2] works for byte arrays and QByteArray, so it would be more generic and allow both kind of objects to be used in case it's neccesary in the future. So... Do you prefer me to change that to startsWith or should I keep it as it is ?